test/test_torch.py

```diff
@@ -12179,6 +12179,23 @@ def test_python_types(self):
         c2 = torch.tensor([True, False], dtype=bool)
         self.assertEqual(c1.dtype, c2.dtype)
 
+    def test_fill_diagonal(self):
```

Comment:
Please add test cases for:

1) A 3-d or larger tensor
2) A non-contiguous tensor